pkgs/applications/video/pipe-viewer/default.nix

```diff
@@ -0,0 +1,94 @@
+{ buildPerlModule, lib, fetchFromGitHub, makeWrapper, wrapGAppsHook
+, withGtk3 ? false
+, ffmpeg
+, gtk3
+, wget
+, xdg-utils
+, youtube-dl
+# Perl dependencies
+, AnyURIEscape
+, DataDump
+, Encode
+, FilePath
+, FileShareDir
+, GetoptLong
+, Gtk3
+, HTTPMessage
+, JSON
+, JSONXS
+, LWPProtocolHttps
+, LWPUserAgentCached
+, Memoize
+, PathTools
+, ScalarListUtils
+, TermReadLineGnu
+, TestPod
+, TextParsewords
+, UnicodeLineBreak
+}:
+
+buildPerlModule rec {
+  pname = "pipe-viewer";
+  version = "0.1.3";
+
+  src = fetchFromGitHub {
+    owner = "trizen";
+    repo = "pipe-viewer";
+    rev = version;
+    hash = "sha256-XoTOjUzhJD3Q0sENe9m5etqbAfBh83B/xmRmqz8zETU=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ] ++
+    lib.optionals withGtk3 [ wrapGAppsHook ];
+
+  buildInputs = [
+    AnyURIEscape
+    DataDump
+    Encode
+    FilePath
+    GetoptLong
+    HTTPMessage
+    JSON
+    JSONXS
+    LWPProtocolHttps
+    LWPUserAgentCached
+    Memoize
+    PathTools
+    ScalarListUtils
+    TermReadLineGnu
+    TextParsewords
+    UnicodeLineBreak
+  ] ++ lib.optionals withGtk3 [
+    FileShareDir
+    Gtk3
+  ];
+
+  # Not supported by buildPerlModule
+  # and the Perl code fails anyway
+  # when Getopt::Long sets $gtk in Build.PL:
+  # Modification of a read-only value attempted at /nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-perl5.34.0-Getopt-Long-2.52/lib/perl5/site_perl/5.34.0/Getopt/Long.pm line 585.
+  #buildFlags = lib.optional withGtk3 "--gtk3";
+  postPatch = lib.optionalString withGtk3 ''
+    sed -i 's|my $gtk .*|my $gtk = 1;|' Build.PL
+  '';
+
+  checkInputs = [
+    TestPod
+  ];
+
+  postInstall = ''
+    wrapProgram "$out/bin/pipe-viewer" \
+      --prefix PATH : "${lib.makeBinPath [ ffmpeg wget youtube-dl ]}"
+  '' + lib.optionalString withGtk3 ''
+    wrapProgram "$out/bin/gtk-pipe-viewer" \
+      --prefix PATH : "${lib.makeBinPath [ ffmpeg wget xdg-utils youtube-dl ]}"
```

Comment:
Does this double wrap?

Reply:
@SuperSandro2000, good catch, it was even a triple wrap for `gtk-pipe-viewer`, I've fixed this by setting `dontWrapGApps` and passing `gappsWrapperArgs` to the `wrapProgram` on `gtk-pipre-viewer`, thanks!